src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        while (true) {
+            raw_in = sc.nextLine().trim();
+            input = raw_in.split(" ");
+            System.out.println(h_rule);
+            String joined = "";
+            String timing = "";
+            if (raw_in.equals("bye")) {
+                System.out.println("Goodbye! Have a nice day!\n");
+                break;
+            } else {
+                switch (input[0]) {
+                    case "list":
+                        int index = 1;
+                        for (Task s : itemList) {
+                            System.out.println(index + ". " + s.toString());
+                            index++;
+                        }
+                        break;
+                    case "done":
+                        int itemToBeUpdatedIndex = Integer.parseInt(input[1]) - 1;
+                        Task task = itemList.get(itemToBeUpdatedIndex);
+                        task.markDone();
+                        fileContents.set(itemToBeUpdatedIndex, task.toString());
+                        System.out.println("Alright, I will mark this as done.\n" + input[1] + ". " + task.toString());
+                        break;
+                    case "todo":
+                        for (int i = 1; i < input.length; i++) {
+                            joined = joined + " " + input[i];
+                        }
+                        itemList.add(new Todo(joined));
+                        fileContents.add(new Todo(joined).toString());
+                        count++;
+                        System.out.println("Added " + raw_in + "\nYou now have " + count + " items in your list.");
+                        break;
+                    case "deadline":
+                        int seq = 0;
+                        while (!input[seq].equals("/by")) {
+                            joined = joined + " " + input[seq];
+                            seq++;
+                            if (seq == input.length) {
+                                throw new DukeDescriptionException(input[0]);
+                            }
+                        }
+
+                        if (seq + 1 == input.length) {
+                            throw new DukeTimingException(input[0]);
+                        }
+
+                        for (int i = seq + 1; i < input.length; i++) {
+                            timing = timing + " " + input[i];
+                        }
+
+                        itemList.add(new Deadline(joined, timing.trim()));
+                        fileContents.add(new Deadline(joined, timing.trim()).toString());
+                        count++;
+                        System.out.println("Added " + joined + "\nYou now have " + count + " items in your list.");
+                        break;
+                    case "event":
+                        int seq2 = 0;
```

Comment:
Perhaps "seq2" here can be renamed to secondSequence as it would make it easier to distinguish from the above statements.